### PR TITLE
Nix run #155

### PR DESCRIPTION
**Pull request by dmadisetti**
It does feel like this code is pretty extensive. introducing a flake could be a nice way to get things under wraps with some tests. I added one just for fun.

- Exposed cutom-nix derivation for remote runs
- Added basic unit test
- Updated docs
-  Added result to gitignore

Less minimal than I hoped, but still pretty small

### CODE REVIEW

**Comment by dmadisetti** [comment]
Whitespace changes are just from my formatter, I can revert them if you want.

**Comment by dmadisetti** [comment]
You can try with `nix run github:anothergroupchat/molten-nvim/nix-run-115#`
and run the unit test with `nix flake check github:anothergroupchat/molten-nvim/nix-run-115#`

If you want to make a github action for CI, see https://github.com/DeterminateSystems/nix-installer?tab=readme-ov-file#as-a-github-action

**Comment by benlubas** [comment]
this is really cool, I'll check it out. Thanks for the contribution!

**Comment by benlubas** [comment]
The demo is really cool. A few thoughts:
- image rendering isn't enabled--need to set `vim.g.molten_image_provider = "image.nvim"` for that to work
- The user's terminal has to support Kitty Graphics, we should probably mention that in the NixOS doc). idk if we can support Uberzug++ with a command line arg or flag?
- I wonder if it's worth while to add a short demo file that's automatically opened so the user can follow some type of tutorial
- The user still has to setup a Jupyter Kernel themselves as far as I can tell. Do you know if we could temporarily install an ipython kernel that molten can find? That would make the demo a lot smoother for people who are just trying this out
- While we're not using nightly, it's probably worth it to include a color scheme, as currently `:MoltenInfo` is _very_ purple and hard to read

---

I'm not sure how useful the checks are. It's nice to make sure the entire plugin hasn't crashed and burned I guess. I'm not opposed to keeping it around.

Do people normally use checks for lint rules/formatters too? or are they more for integration type tests?

**Comment by dmadisetti** [comment]
Sure, I'll try to set it up with images. You can have nixpkgs provide kitty + have other packages of just molten

Hmm, I had the python kernel come up with `:MoltenInit`? I'm not sure what you mean

Checks are set to run unit tests and basically check package integrity. I think they are implicitly added for some packaging like rust. Might help keep all the various features on track

**Comment by benlubas** [comment]
> I had the python kernel come up with :MoltenInit

Oh interesting. I only saw the kernels that I've already got installed on my system.

File: flake.nix [hunk 1]
```diff
@@ -6,31 +6,103 @@
...
+                   +q \
+                   2>&1 | grep "\[Molten\] Kernel 'python3' (id: python3) is ready." || exit 1
+              echo OK > $out
+            '';
+          };
+        };
 
+        packages = rec {
+          molten = pkgs.writeShellScriptBin "molten" ''
+            PATH=$PATH:${pkgs.python3Packages.jupytext}/bin ${neovim-custom}/bin/nvim \
```
Comment (benlubas) [inline flake.nix:97]: Also just saw this. I don't think that jupytext needs to be in path here. Molten doesn't use jupytext at all.

**Comment by benlubas** [comment]
I'm going to close this for inactivity, feel free to open it back up if you have time to work on it a bit more

**Pull request closed without merging**